clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -573,6 +575,10 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                 } else {
                     future.raise(error);
                 }
+            } else if (error == Errors.UNSUPPORTED_VERSION) {
+                log.error("Attempt to join group failed due to unsupported version error. Please unset field group.instance.id and retry" +
```

Comment:
nit: missing space after "retry"

I think it would be better to just say that the broker does not support the use of `group.instance.id`. It's a little weird to tell the user to unset a configuration.

Reply:
Got it